### PR TITLE
The unsupported_fp16_list using in AMP will be created automatically during the runtime.

### PR DESCRIPTION
**Pull request by wzzju**
### PR types

Function optimization

### PR changes

Others
### Describe

After this pull request, the `unsupported_fp16_list` using in AMP will be created automatically during the runtime.

### CODE REVIEW

File: paddle/fluid/pybind/pybind.cc [hunk 1]
```diff
@@ -181,6 +184,53 @@ bool SupportsBfloat16FastPerformance() {
 #endif
 }
 
+std::tuple<std::unordered_set<std::string>, std::unordered_set<std::string>,
```
Comment (zhhsplendid) [inline paddle/fluid/pybind/pybind.cc:197]: Something for the readability, please add simple comment.

The function checks all ops supports input place and dtype, returns all_ops, supported_ops, and unsupported_ops. The function name and return type are not obvious enough for above information.

File: paddle/fluid/pybind/pybind.cc [hunk 2]
```diff
@@ -181,6 +184,53 @@ bool SupportsBfloat16FastPerformance() {
...
+           std::unordered_set<std::string>>
+OpSupportedInfos(std::string place, framework::proto::VarType::Type dtype) {
```
Comment (zhhsplendid) [inline paddle/fluid/pybind/pybind.cc]: Would you mind input a `const std::string& place` ?

This is just following C++ style:
https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs

If the `std::toupper` causes the trouble for using const reference, you can do a copy inside the function.

File: paddle/fluid/pybind/pybind.cc (hunk 1 above)
Comment (wzzju) [reply paddle/fluid/pybind/pybind.cc:197]: Done.

File: paddle/fluid/pybind/pybind.cc (hunk 2 above)
Comment (wzzju) [reply paddle/fluid/pybind/pybind.cc]: Done.

**Review by zhhsplendid — DISMISSED** [review]
LGTM

**Review by zhhsplendid — DISMISSED** [review]
LGTM

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**